core/genesis.go

```diff
@@ -207,6 +207,9 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *Genesis, override
 	if overrideIstanbul != nil {
 		newcfg.IstanbulBlock = overrideIstanbul
 	}
+	if err := newcfg.CheckConfigForkOrder(); err != nil {
```

Comment:
You should also check the order of config in `genesis.Commit` function.

Reply:
Not sure about that.. the `rawdb.Write..` - methods do not return any errors, so I'd have to introduce that. You're probably right about adding it to `Commit` aswell

Reply:
I mean before the `rawdb.WriteChainConfig` we should check the order of config.

Reply:
I think I implemented what you meant, please check if you agree